Simulator/Recorders/XmlRecorder.cpp

```diff
@@ -137,13 +140,16 @@ string XmlRecorder::toXML(const string &name, vector<multipleTypes> &singleBuffe
       << "\" multiplier=\"1.0\">" << endl;
    os << "   ";
 
+   // Retrives value from variant
    for (const multipleTypes &element : singleBuffer_) {
       if (basicType == "uint64_t") {
          os << get<uint64_t>(element) << " ";
       } else if (basicType == "double") {
-         os << get<double>(element) << " ";
-      } else if (basicType == "string") {
-         os << get<string>(element) << " ";
+         os << get<bool>(element) << " ";
+      } else if (basicType == "int") {
+         os << get<int>(element) << " ";
+      } else if (basicType == "BGFLOAT") {
+         os << get<BGFLOAT>(element) << " ";
```

Comment:
Add one more `else if`, to compare with `"undefined"` and generate an error message.